src/mscorlib/src/Internal/Runtime/Augments/EnvironmentAugments.cs

```diff
@@ -0,0 +1,21 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System;
+
+namespace Internal.Runtime.Augments
+{
+    /// <summary>For internal use only.  Exposes runtime functionality to the Environments implementation in corefx.</summary>
+    public static class EnvironmentAugments
+    {
+        public static int CurrentManagedThreadId => Environment.CurrentManagedThreadId;
+        public static void Exit(int exitCode) => Environment.Exit(exitCode);
+        public static int ExitCode { get { return Environment.ExitCode; } set { Environment.ExitCode = value; } }
+        public static void FailFast(string message, Exception error) => Environment.FailFast(message, error);
+        public static string[] GetCommandLineArgs() => Environment.GetCommandLineArgs();
+        public static bool HasShutdownStarted => Environment.HasShutdownStarted;
+        public static string StackTrace => Environment.StackTrace;
```

Comment:
This will return extra useless lines in the stacktrace based on how the optimizations kicked in. We will need to:
- Disable optimizations on the forwarder in corefx to make sure that there is fixed number of frames to skip
- Skip the given number of frames here by calling StackTrace constructor with `skipFrames` argument, instead of just forwarding to Environment.